cmd/ctl/pkg/version/version.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/jetstack/cert-manager/pkg/util"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/yaml"
+)
+
+// Options is a struct to support version command
+type Options struct {
+	Output string
+	Short  bool
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdVersion returns a cobra command for fetching versions
+func NewCmdVersion(ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print the cert-manager-ctl version",
+		Long:  "Print the cert-manager-ctl version",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "One of 'yaml' or 'json'.")
```

Comment:
You're setting the default value here to `o.Output` - shouldn't this be `yaml` or `json`? It'll be `` (empty) otherwise

Reply:
Ahh, I see, 'empty' means 'human readable'. Maybe just change this to `""` then instead of `o.Short`? :)